src/query/v2/plan/operator.cpp

```diff
@@ -462,6 +463,202 @@ class DistributedScanAllAndFilterCursor : public Cursor {
   std::optional<std::vector<Expression *>> filter_expressions_;
 };
 
+class DistributedScanAllByPrimaryKeyCursor : public Cursor {
+ public:
+  explicit DistributedScanAllByPrimaryKeyCursor(
+      Symbol output_symbol, UniqueCursorPtr input_cursor, const char *op_name,
+      std::optional<storage::v3::LabelId> label,
+      std::optional<std::pair<storage::v3::PropertyId, Expression *>> property_expression_pair,
+      std::optional<std::vector<Expression *>> filter_expressions, std::optional<std::vector<Expression *>> primary_key)
+      : output_symbol_(output_symbol),
+        input_cursor_(std::move(input_cursor)),
+        op_name_(op_name),
+        label_(label),
+        property_expression_pair_(property_expression_pair),
+        filter_expressions_(filter_expressions),
+        primary_key_(primary_key) {
+    ResetExecutionState();
+  }
+
+  enum class State : int8_t { INITIALIZING, COMPLETED };
+
+  using VertexAccessor = accessors::VertexAccessor;
+
+  bool MakeRequest(RequestRouterInterface &request_router, ExecutionContext &context) {
+    {
+      SCOPED_REQUEST_WAIT_PROFILE;
+      std::optional<std::string> request_label = std::nullopt;
+      if (label_.has_value()) {
+        request_label = request_router.LabelToName(*label_);
+      }
+      current_batch_ = request_router.ScanVertices(request_label);
+    }
+    current_vertex_it_ = current_batch_.begin();
+    request_state_ = State::COMPLETED;
+    return !current_batch_.empty();
+  }
+
+  bool MakeRequestSingleFrame(Frame &frame, RequestRouterInterface &request_router, ExecutionContext &context) {
+    {
+      SCOPED_REQUEST_WAIT_PROFILE;
```

Comment:
First of all, I think this `SCOPED_REQUEST_WAIT_PROFILE` is not the best solution, because it is hard to put it on a separate scope and return something from the request. However for now we can use something like this:

```
      auto get_prop_result = std::invoke([&context, &req]() mutable {
        SCOPED_REQUEST_WAIT_PROFILE;
        return request_router.GetProperties(req);
      });
```